src/spikeinterface/extractors/neoextractors/neobaseextractor.py

```diff
@@ -293,35 +294,125 @@ def __init__(self, block_index=None, sampling_frequency=None, use_natural_unit_i
                 use_natural_unit_ids=self.use_natural_unit_ids,
                 t_start=t_start,
                 sampling_frequency=sampling_frequency,
-                handle_spike_frame_directly=self.handle_spike_frame_directly,
+                neo_returns_timestamps=self.neo_returns_timestamps,
             )
 
             self.add_sorting_segment(sorting_segment)
 
-    def _auto_guess_sampling_frequency(self):
+    def _get_t_start_from_signal_stream(self, segment_index):
+        # Get the stream index using the sampling frequency
+        sampling_frequency = self._sampling_frequency
+        stream_id_to_sampling_frequencies = self.build_stream_id_to_sampling_frequency_dict()
+
+        streams_with_given_sampling_frequency = [
+            stream_id for stream_id, sf in stream_id_to_sampling_frequencies.items() if sf == sampling_frequency
+        ]
+
+        # Get the stream index for each stream in streams_with_given_sampling_frequency
+        stream_id_to_stream_index = dict()
+        for stream_id in streams_with_given_sampling_frequency:
+            signal_streams = self.neo_reader.header["signal_streams"]
+            stream_index = (signal_streams["id"] == stream_id).nonzero()[0][0]
+            stream_id_to_stream_index[stream_id] = stream_index
+
+        # Get a dictionary from stream_index to t_start
+        stream_id_to_t_start = dict()
+        for stream_id, stream_index in stream_id_to_stream_index.items():
+            t_start = self.neo_reader.get_signal_t_start(
+                block_index=self.block_index,
+                seg_index=segment_index,
+                stream_index=stream_index,
+            )
+            stream_id_to_t_start[stream_id] = t_start
+
+        available_t_starts = list(stream_id_to_t_start.values())
+        unique_t_starts = set(available_t_starts)
+
+        if len(unique_t_starts) == 1:
+            t_start = available_t_starts[0]
+
+        else:
+            error_msg = (
```

Comment:
Here:
@alejoe91 @samuelgarcia 
What if there is more than one t_start corresponding to the chosen sampling frequency?